spec/integration/system_pager_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+RSpec.describe TTY::Pager::SystemPager do
+  around :each do |example|
+    Dir.mktmpdir do |dir|
+      FileUtils.cd(dir) do
+        example.run
+      end
+    end
+  end
+
+  it "paginates some text" do
+    output_path = Pathname.new('output.txt')
+
+    described_class.page("Some text", command: "cat > #{output_path}")
+    expect(output_path.read).to eq "Some text"
+  end
+
+  it "allows pagination to happen asynchronously" do
+    output_path = Pathname.new('output.txt')
+    File.open('external_pager.rb', 'w') do |script|
+      script.write(<<~RUBY)
+        File.open('output.txt', 'w') do |output|
+          ARGF.each { |line| output.puts(line) }
+        end
+      RUBY
+    end
+
+    described_class.page(command: "ruby external_pager.rb") do |pager|
+      pager.puts("one")
+      pager.write("two\n")
+      pager.try_write("three\n")
+    end
+
+    expect(output_path.read).to eq "one\ntwo\nthree\n"
+  end
+
+  it "stops paginating once external tool is closed" do
+    output_path = Pathname.new('output.txt')
+    File.open('external_pager.rb', 'w') do |script|
+      script.write(<<~RUBY)
+        ARGF.each_with_index do |line, index|
+          File.open('output.txt', 'a') do |output|
+            output.puts(line)
+            output.flush
+          end
+        end
+      RUBY
```

Comment:
I kind of feel this could be placed in a `spec/fixtures/eternal_pager.rb` and referenced with a helper `fixtures_path("external_pager.rb")`.  Then the command could be invoked similar to `cat.rb`, namely, `"ruby #{fxitures_path....} > output.txt"`. What do you reckon?

Reply:
I've extracted it to a separate file, but weirdly, if I use `> output_file` instead of providing and opening it as an argument, I get strange errors like:

```
Traceback (most recent call last):
-:1:in `<main>': undefined local variable or method `one' for main:Object (NameError)
```

I have no good explanation as to why the contents of the file somehow get evaluated as ruby. The current setup works for me, though, so I can live with not having a reasonable explanation. Let me know what you think.